### ISSUE TITLE
[Bug]: Accessibility properties SizeOfSet and PositionOfSet are not being set when using <ReactiveWindow>

### ISSUE THREAD

**Issue by hicks86** [reactiveui/ReactiveUI#3637]
### Describe the bug 🐞

As a part of our development process we run [Microsoft's Accessibility Insights FastPass tool](https://accessibilityinsights.io/docs/windows/getstarted/fastpass/) to ensure that UI changes are meeting this accessibility standard.

There is one issue that we cannot seem to resolve, and it seems to be linked to someting weird going on behind the scenes of ReactiveUI (from what I can tell). 
When binding a list to a ComboBox using ReactiveUI there seems to be an issue that is being flagged by the tool around the values of the `AutomationProperties.SizeOfSet` and `AutomationProperties.PositionOfSet`. As of .NET48 these values should be automatically filled in in a WPF application. These values are used by a screenreader to inform the user where they are on a particular page (e.g. position 4 of 10).
Anyways when binding a list to a ComboBox using a regular WPF code behind binding (and just using <Window> instead of <ReactiveWindo>) this is no longer an issue on the accessibility tool.

### Step to reproduce

1. Go to the reproduction and start the application
2. Using  [Microsoft's Accessibility Insights FastPass tool](https://accessibilityinsights.io/docs/windows/getstarted/fastpass/) run a test on the combobox.
3. You will see the accessibility issue ** "The elements ControlType requires valid values for SizeOfSet and PositionOfSet"**
4. Then swap to use a classic WPF code-behind approach to binding list to a combobox (Swap `<ReactiveWindow>` for `<Window>`,  uncomment the `WpfCombobox.ItemSource = ...` code and comment out the ReactuveUI setup).
5. Run the FastPass tool again. 
6. You will see there are no accessibility issues.

### Reproduction repository

https://github.com/hicks86/ReactiveUIAccessibilityRepro

### Expected behavior

When you run the accessibility tool there should be no accessibility issue present for the SizeOfSet and PositionOfSet AutomationProperties.

### IDE

Visual Studio 2022

### Operating system

Windows 10

### ReactiveUI Version

19.4.1

### Additional information ℹ️

Happy to provide any other information needed. It would be good to get to the bottom of this.

**Comment by hicks86** [comment]
Anyone had a chance to look at this? Or provide some comments? Cheers

**Comment by ChrisPulman** [comment]
I have been travelling for the past few weeks but should have some time available this weekend to take a look at this and some of the other recent observations raised. 
Thank you for your patience.

**Comment by hicks86** [comment]
Thank you Chris.

**Comment by glennawatson** [comment]
I have a strong suspicion it's the way we handle `ItemsSource` overwriting the way these automatic properties are set. 

https://github.com/reactiveui/ReactiveUI/blob/0a61f2ab88d3a7e0721c1dc41fed43f30900c011/src/ReactiveUI.Wpf/Common/AutoDataTemplateBindingHook.cs#L43

When doing binding we will use a IPropertyBindingHook to look up from the view resolution and find the appropriate value to show. 

I'm going to test this weekend and override the IPropertyBindingHook in the splat registrations with a custom one to see if its related.

**Comment by ChrisPulman** [comment]
Please use [ReactiveUI.SourceGenerators](https://github.com/reactiveui/ReactiveUI.SourceGenerators#usage-iviewfor-iviewfornameofviewmodelname) `[IViewFor<ViewModel>()]` instead of `ReactiveWindow<ViewModel>` to resolve this issue.
